### PR TITLE
Inject sleep during etcd bootstrap to reproduce etcd-io#16666 

### PR DESCRIPTION
**Pull request by serathius**
Part of https://github.com/etcd-io/etcd/issues/16673

This is enough to reproduce https://github.com/etcd-io/etcd/issues/16666

### CODE REVIEW

**Comment by serathius** [comment]
cc @ahrtr

**Review by ahrtr — APPROVED** [review]

File: tests/robustness/failpoints.go [hunk 1]
```diff
@@ -527,3 +529,48 @@ func (f delayPeerNetworkFailpoint) Name() string {
...
+
+func (f killAndGofailSleep) Inject(ctx context.Context, t *testing.T, lg *zap.Logger, clus *e2e.EtcdProcessCluster) error {
+	member := clus.Procs[rand.Int()%len(clus.Procs)]
+	for member.IsRunning() {
+		err := member.Kill()
+		if err != nil {
+			lg.Info("Sending kill signal failed", zap.Error(err))
+		}
+		err = member.Wait(ctx)
+		if err != nil && !strings.Contains(err.Error(), "unexpected exit code") {
```
Comment (fuweid) [inline tests/robustness/failpoints.go:546]: The `Wait` seems to return `ctx.Err` timeout error only.

**Review by fuweid — APPROVED** [review]
LGTM

File: tests/robustness/failpoints.go (hunk 1 above)
Comment (serathius) [reply tests/robustness/failpoints.go:546]: Copied from killFailpoint. Handling "unexpected exit code" comes from https://github.com/etcd-io/etcd/pull/14672

**Pull request merged**